versions/3.0.md

```diff
@@ -1123,6 +1125,344 @@ application/json:
+parameters  | Link Parameters Object | an Object representing parameters to pass to an operation as specified with `operationId` or identified via `href`.
+description | string | a description of the link, supports GFM.
+
+#### <a name="responsePayload"></a>Response Payload Values
+
+Payload values are only available in parseable response payloads which match the advertised media-type.  In all cases, if a value does _not_ exist, the parameter will be considered a `null` value (as opposed to an empty value) and _not_ passed as a parameter to the linked resource. In cases where a value is required, and a parameter is not supplied, the client _may_ choose to not follow the link definition. 
+
+### Example
+
+Response payload:
+```json
+{
+    "id": "df71a505-07bc-458a-a5c0-73c0340d1ec7",
+    "firstname": "Ash",
+    "lastname": "Williams"
+}
+```
+
+Payload Variables:
+```yaml
+id: df71a505-07bc-458a-a5c0-73c0340d1ec7
+firstname: Ash
+lastname: Williams
+missingValue: null
+```
+
+In situations where variables appear in an array, an array of variables will be extracted.  For example:
+
+```json
+[
+    { "color": "red" },
+    { "color": "green" },
+    { "color": "blue" }
+]
+```
+
+will be extracted as such:
+
+```json
+color: ["red", "green", "blue"]
+```
+
+The variables generated can be used in locations prescribed by the definition.
+
+
+### Variable substitution
+In all cases, _variables_ identified in the response payload may be substitued **only** in locations prescribed in the link definitions.  All such locations are prefixed by a `$response` keyword and surrounded by curly brackets `{}`. 
+
+### Example
+Computing a link from a response object is performed as follows:
+
+```yaml
+Addresses:
+  href: /users/{$response.id}/address
+```
+
+Where the `$response.id` from the example above would yield the target:
+
+```yaml
+href: '/users/df71a505-07bc-458a-a5c0-73c0340d1ec7/address'
+```
+
+And the array example:
+
+```yaml
+ColorSelection:
+  href: 'http://colors.my-server.com/colors/{$response.color}'
+```
+
+Would produce the following links:
+
+```yaml
+href: 'http://colors.my-server.com/colors/red'
+href: 'http://colors.my-server.com/colors/green'
+href: 'http://colors.my-server.com/colors/blue'
+```
+
+As with all links, it at the the clients' discretion to follow them, and permissions and the existence of a value is not guaranteed soley by the existence of a link relationship. [**I assume you mean that the existance of the link relationship does not guarantee a successful response from the call**]
+
+
+### Example
+
+The example below shows how relationships in the BitBucket API can be represented with the proposed OAI link scheme.  This example uses `operationId` values to link responses to possible operations.
```

Comment:
Note, this is a very long example and should probably be shortened for the documentation.